chia/wallet/nft_wallet/uncurry_nft.py

```diff
@@ -87,9 +89,17 @@ class UncurriedNFT(Streamable):
     """
     royalty_address: Optional[bytes32]
     trade_price_percentage: Optional[uint16]
+    coin_state: Optional[CoinState] = None
```

Comment:
Not necessarily requesting this but any thoughts about using some extra class here like in the CAT case instead of starting to mess with `UncurriedNFT`? 

```python
@streamable
@dataclass(frozen=True)
class NFTCoinData(Streamable):
    uncurried_nft: UncurriedNFT
    parent_coin_state: CoinState
    parent_coin_spend: CoinSpend
```

Then you could drop the streamable additions again from `UncurriedNFT` and the state/spend would not need to be optional.

Reply:
I think we still need the Streamable annotation since the members of a Streamable have to be Streamable, too, right? I think use a unified naming convention is a good idea.

Reply:
> I think we still need the Streamable annotation since the members of a Streamable have to be Streamable, too, right? I think using a unified naming convention is a good idea. I will change this.

Oh yeah your right, the streamable is still required then!